### PR TITLE
Add support for ATP protocol to QML

### PR DESCRIPTION
**Pull request by AlexanderOtavka**
Register a custom QNetworkAccessManager with the QML engine to handle ATP urls.

Fixes [FogBugz 1023](https://highfidelity.fogbugz.com/f/cases/1023/QML-Windows-10)

### Testing
1. Upload a file called "ExampleWindow.qml" to the asset server root folder (see below for contents)
2. Upload an image called "image-asset.jpg" (any JPG image off google should do)
3. Copy-paste the JavaScript below into the console
4. A window should appear with the previously uploaded image in it
5. Try changing the url in ExampleWindow.qml to `/image-asset.jpg`, then to `atp:/image-asset.jpg`, making sure to reupload ExampleWindow.qml and restart interface each time

ExampleWindow.qml:
```
import QtQuick 2.3

Image {
    width: 400;
    height: 150;
    fillMode: Image.PreserveAspectFit;
    source: "image-asset.jpg";
}
```

javascript code:
```
var myWindow = new OverlayWindow({
    title: "My Window",
    source: "atp:/exampleWindow.qml",
    width: 400,
    height: 150,
    visible: true
});
```

### CODE REVIEW

File: libraries/networking/src/QmlAtpReply.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
+//
+//  QmlAtpReply.cpp
```
Comment (jherico) [inline libraries/networking/src/QmlAtpReply.cpp]: should just be called `AtpReply`

File: libraries/networking/src/QmlNetworkAccessManager.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#include <QtNetwork/QNetworkRequest>
+#include <QtQml/QQmlNetworkAccessManagerFactory>
+
+class QmlNetworkAccessManagerFactory : public QQmlNetworkAccessManagerFactory {
+public:
+    QNetworkAccessManager* create(QObject* parent);
+};
+
+
+class QmlNetworkAccessManager : public QNetworkAccessManager {
```
Comment (jherico) [inline libraries/networking/src/QmlNetworkAccessManager.h]: should probably just be called `HifiNetworkAccessManager`

File: libraries/networking/src/QmlNetworkAccessManager.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//
+
+#ifndef hifi_QmlNetworkAccessManager_h
+#define hifi_QmlNetworkAccessManager_h
+
+#include <QtNetwork/QNetworkAccessManager>
+#include <QtNetwork/QNetworkRequest>
+#include <QtQml/QQmlNetworkAccessManagerFactory>
+
+class QmlNetworkAccessManagerFactory : public QQmlNetworkAccessManagerFactory {
```
Comment (jherico) [inline libraries/networking/src/QmlNetworkAccessManager.h]: This should be moved to a file next to the OffscreenQmlSurface class in the GL library.

**Comment by hifi-gustavo** [comment]
Build console output for commit: cb153f2a29f59297fa8566918997a07335f741fb available at: https://gist.github.com/5aac235584efadc4597f2902f255fe87<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-cb153f2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-cb153f2.dmg

**Comment by jherico** [comment]
The low level networking code isn't QML specific and shouldn't use QML in the name.  The only thing that should have QML in the name is the class that derives from `QQmlNetworkAccessManagerFactory`

File: libraries/networking/src/QmlNetworkAccessManager.h (hunk 2 above)
Comment (AlexanderOtavka) [inline libraries/networking/src/QmlNetworkAccessManager.h]: There is already a NetworkAccessManager used elsewhere.  I wasn't sure if we wanted to roll this ATP special casing into that one, so I named it QmlNetworkAccessManager to show that it's only used by the QML engine.  I could merge this with NetworkAccessManager if you think that's a good idea.

File: libraries/networking/src/QmlAtpReply.cpp (hunk 1 above)
Comment (AlexanderOtavka) [inline libraries/networking/src/QmlAtpReply.cpp]: See my other comment, but yeah, I could do that if you think that kind of generalization is what we want.

**Comment by AlexanderOtavka** [comment]
@jherico Ok, will do.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 51fb52977e039af704dfc74b7faf653ed80a6d09 available at: https://gist.github.com/5aac235584efadc4597f2902f255fe87<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-51fb529.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-51fb529.dmg

**Comment by AlexanderOtavka** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 51fb52977e039af704dfc74b7faf653ed80a6d09 available at: https://gist.github.com/5aac235584efadc4597f2902f255fe87<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-51fb529.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-51fb529.dmg

**Comment by AlexanderOtavka** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 51fb52977e039af704dfc74b7faf653ed80a6d09 available at: https://gist.github.com/5aac235584efadc4597f2902f255fe87<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-51fb529.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8391-51fb529.dmg

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**QA Pass**: Success
**Additional Test Results + Observations**:  Success.  Moved qml file to asset root dir and created jpg and ran JS in console.  Image appeared as expected.

**Pull request merged**